Source/ARTDataEncoder.m

```diff
@@ -42,10 +47,23 @@ - (instancetype)initWithCipherParams:(ARTCipherParams *)params error:(NSError **
                 return nil;
             }
         }
+        Class vcdiffPluginClass = [plugins pluginClassOf:ARTPluginTypeVCDiff];
+        if ([vcdiffPluginClass conformsToProtocol:@protocol(ARTVCDiffDecoder)]) {
+            _vcdiffDecoder = [[vcdiffPluginClass alloc] init];
+        }
     }
     return self;
 }
 
+- (void)setLastMessageData:(nullable id)data {
```

Comment:
@QuintinWillison Well, I didn't use the `applyDelta:deltaId:baseId:error:` because the `ARTVCDiffDecoder` interface has only the `decode(data, deltaBase)` method. Should I change the interface?

Reply:
I was referring to the available interfaces in [the delta codec](https://github.com/ably/delta-codec-cocoa)... if you trace it all the way back there then that implementation should provide support for base storage, id tracking, etc..

Or am I being stupid and/or missing something? 😕

Reply:
@QuintinWillison My guide was the Java implementation and I used the same interface defined in Java https://github.com/ably/ably-java/blob/0415135d9a1aaf6fbd520458e60dd0d704860498/lib/src/main/java/io/ably/lib/types/VCDiffDecoderHelper.java#L11

Even so, I'll update the code 👍

Reply:
Fair enough. I didn't refer to the Java code alongside. @codemerx, your thoughts on this please? I didn't spot that helper when I looked at the Java PR. I'm a bit confused, TBH.

Reply:
Did the change baf90a2.

Reply:
I am not really sure I have all teh background to understand what this discussion is about. The `VCDiffDecoderelper` is a convenience wrapper of the third party vcdiff decoding lib we use in Java. It is not meant to be public API of `ably-java`